### DISCUSSION TITLE
Linux on Chromebook Support

### DISCUSSION THREAD

**Discussion by ghost** [srl-labs/containerlab#1261] [Q&A]
Is Linux on Chromebook supported by containerlab? I tried the lab, /etc/containerlab/lab-examples/srl02/*, but got the following errors:

ERRO[0064] failed to run postdeploy task for node srl1: command execution error:At line 10: Error: Error in path: 
    [InvalidArgument] Unacceptable syslog configuration
Error 'Revert failed':
    [Internal] Failed to revert to pre-commit configuration
Error 'Application failure during commit':
    [FailedPrecondition] Applications have failed: log_mgr
 
ERRO[0071] failed to run postdeploy task for node srl2: command execution error:At line 10: Error: Error in path: 
    [InvalidArgument] Unacceptable syslog configuration
Error 'Revert failed':
    [Internal] Failed to revert to pre-commit configuration
Error 'Application failure during commit':
    [FailedPrecondition] Applications have failed: log_mgr

I also set up the same lab on WSL and got no such error. Another symptom I observed was the Chromebook lab allowed me to ssh with admin/admin while the WSL lab allowed admin/NokiaSrl1!

**Comment by hellt** [comment]
Hi. Unfortunately I don't have a Chromebook to test and noone before revealed they run clab on Chromebooks.

Looks like it doesn't like/allow syslog modifications that srl container needs. Is there a workaround I can't say without having a similar system to test.

**Comment by ghost** [comment]
Hi. If I clab destroy but leave the folder/files that were created and deploy again, I get a different error. Still seems like a modifications issue.

ERRO[0012] failed to run postdeploy task for node srl2: command execution error:At line 1: Error: Unable to modify configuration or use tools commands during yang reload 
ERRO[0012] failed to run postdeploy task for node srl1: command execution error:At line 1: Error: Unable to modify configuration or use tools commands during yang reload

**Comment by ghost** [comment]
Looking through the docker container logs, I think I found the culprit but not sure I can get the permissions to change it without dev mode. I tried setting the values in /etc/sysctl.conf of Debian but they won't take.

sysctl: permission denied on key 'fs.pipe-max-size'
sysctl: permission denied on key 'fs.pipe-user-pages-soft'
/opt/srlinux/bin/sr_linux: line 278: ulimit: open files: cannot modify limit: Operation not permitted

**Comment by hellt** [comment]
It looks like srlinux is not compatible with your kernel/os settings.
You can tinker with the way sysctls are set in containerlab code here https://github.com/srl-labs/containerlab/blob/main/nodes/srl/srl.go#L70

**Reply by ghost** [reply]
Chrome OS's Debian runs in lxc under a crosvm. I could not figure out how to get permissions for sysctl in Debain. Looks like crosvm is pretty locked down.

**Comment by ghost** [comment]
I ended up running a vm in the Debian container of Chrome OS in order to get around the sysctl permissions. Doesn't feel like to much of a performance hit but we'll see.
